src/main/java/com/google/firebase/database/snapshot/BigDecimalNode.java

```diff
@@ -0,0 +1,60 @@
+package com.google.firebase.database.snapshot;
+
+import java.math.BigDecimal;
+
+public class BigDecimalNode extends LeafNode<BigDecimalNode> {
+
+  private final String value;
+
+  public BigDecimalNode(BigDecimal value, Node priority) {
+    this(value == null ? null : value.toString(), priority);
+  }
+
+  BigDecimalNode(String value, Node priority) {
+    super(priority);
+    this.value = value;
+  }
+
+  @Override
+  public Node updatePriority(Node priority) {
+    return new BigDecimalNode(this.value, priority);
+  }
+
+  @Override
+  public Object getValue() {
+    return this.value;
```

Comment:
Should this return a `BigDecimal`? You can reconstruct the original value from the string.